src/app/mini-games/sprint/sprint-mini-game.component.ts

```diff
@@ -1,8 +1,221 @@
+      this.game.words.push(...words);
+    });
+    this.game.gameState = GameState.READY;
+  }
+
+  getReadyToPlay() {
+    this.isStarterCounterShown = false;
+
+    const playGame = () => {
+      if (this.game.words.length) {
+        this.game.gameState = GameState.PLAY;
+        this.nextTurn();
+      } else {
+        this.loadingWordsMessage = 'Слова загружаются...';
+        setTimeout(() => (this.loadingWordsMessage = ''), 1000);
+        setTimeout(playGame, 2000);
+      }
+    };
+
+    playGame();
+  }
+
+  handleTurn(answer: string) {
+    this.game.trainedWordsByIndexes.push(this.game.wordIndex);
+
+    const trainedWord = {
+      id: this.game.words[this.game.wordIndex].id,
+      word: this.game.word,
+      translation: this.game.wordTranslation,
+      timeStamp: Date.now(),
+    };
+
+    if (
+      (answer === Answers.CORRECT && this.game.isTranslationCorrect) ||
+      (answer === Answers.WRONG && !this.game.isTranslationCorrect)
+    ) {
+      this.game.streak++;
+      this.game.multiplier = getPointsMultiplier(this.game.streak);
+      this.game.points += this.game.basePoints * this.game.multiplier;
+      this.game.trainedWords.push({
+        ...trainedWord,
+        result: Answers.CORRECT,
+      });
+
+      this.launchRipple(Answers.CORRECT);
+
+      if (!this.game.isMuted) {
+        if (this.game.streak % 4 === 0 && this.game.streak <= StreakLevel.THIRD) {
+          playSound(Sound.LEVELUP);
+        } else {
+          playSound(Sound.CORRECT);
+        }
+      }
+    } else {
+      this.game.streak = 0;
+      this.game.multiplier = 1;
+      this.game.trainedWords.push({
+        ...trainedWord,
+        result: Answers.WRONG,
+        translation: this.game.words[this.game.wordIndex].wordTranslate,
+      });
+
+      this.launchRipple(Answers.WRONG);
+
+      if (!this.game.isMuted) playSound(Sound.WRONG);
+    }
+    this.nextTurn();
+  }
+
+  nextTurn() {
+    if (this.game.trainedWords.length === MAX_TRAINED_WORDS) {
+      this.gameOver();
+      return;
+    }
+
+    do {
+      this.game.wordIndex = getRandomNumber(this.game.words.length);
+    } while (this.game.trainedWordsByIndexes.includes(this.game.wordIndex));
+
+    this.game.word = this.game.words[this.game.wordIndex].word;
+
+    if (isTranslationCorrect()) {
+      this.game.isTranslationCorrect = true;
+      this.game.wordTranslation = this.game.words[this.game.wordIndex].wordTranslate;
+    } else {
+      this.game.isTranslationCorrect = false;
+      let anotherWordIndex = getRandomNumber(this.game.words.length);
+
+      while (anotherWordIndex === this.game.wordIndex) {
+        anotherWordIndex = getRandomNumber(this.game.words.length);
+      }
+      this.game.wordTranslation = this.game.words[anotherWordIndex].wordTranslate;
+    }
+  }
+
+  pronounceWord(id?: string) {
+    const wordId = id ? id : this.game.words[this.game.wordIndex].id;
```

Comment:
@Suggestion
```const wordId = id || this.game.words[this.game.wordIndex].id;```

Reply:
Changed to: 
const wordId = id ?? this.game.words[this.game.wordIndex].id;